### PR TITLE
Document use of git submodule commands

### PR DESCRIPTION
**Pull request by wolf31o2**
This adds a section on using Git submodules with this repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,37 @@ CDAP should be built using
...
-## Compiling/Installing Apache Sentry 1.7.0
+## Submodules and Git
+
+This repository uses Git submodules to provide links to other
+repositories. Cloning this repository with `--recursive` will
+automatically initialize and checkout the additional repositories.
+These additional repositories are configured to track the correct
+remote branches for a given CDAP release. This repository will have
+the correct Git references stored for a particular tag. Checking
+out a tag to your working directory will update the submodule git
```
Comment (awholegunch) [inline README.md]: submodule Git

File: README.md [hunk 2]
```diff
@@ -10,7 +10,37 @@ CDAP should be built using
...
+references to the ones used to build that tag.
+
+If you cloned without using `--recursive`, you will need to
+initialize and checkout the submodules.
```
Comment (awholegunch) [inline README.md]: If these are the commands to run if you *don't* use _recursive_, a colon would help:
```
checkout the submodules:

File: README.md [hunk 3]
```diff
@@ -10,7 +10,37 @@ CDAP should be built using
...
+
+```bash
+git submodule init
+git submodule update
+```
+
+Starting with Git 1.8, it is possible to track remote branches
+in Git submodules. These branches have already been configured
+for each submodule, and the submodules can be updated to the
+head of that branch by appending `--remote` to the update command.
```
Comment (awholegunch) [inline README.md]:
```
to the update command:

File: README.md [hunk 4]
```diff
@@ -10,7 +10,37 @@ CDAP should be built using
...
+
+```bash
+git submodule update --remote
+```
+
+## Building a CDAP release
+
+### Compiling/Installing Apache Sentry 1.7.0
 
 The CDAP Security Extensions requires you to have Apache Sentry 1.7.0
```
Comment (awholegunch) [inline README.md]:
```
The CDAP Security Extensions require that you have the Apache Sentry 1.7.0

**Comment by awholegunch** [comment]
Four minor comments, fix and merge. LGTM.

File: README.md (hunk 4 above)
Comment (wolf31o2) [inline README.md]: I'll change it, but I was considering "CDAP Security Extensions" as a singular proper noun, as it's a single repository. However, I think I'm making up why that makes sense to anyone else.

**Pull request merged**